### DISCUSSION TITLE
fish

### DISCUSSION THREAD

**Discussion by TwelveNight** [Homebrew/discussions#4488] [Linux] [answered]
### Additional Info

- [X] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.16
ORIGIN: https://mirrors.ustc.edu.cn/brew.git
HEAD: 01e0c20d01c6aab28d844280d130f0e745e2ddd9
Last commit: 7 days ago
Core tap JSON: 07 May 12:36 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_BREW_GIT_REMOTE: https://mirrors.ustc.edu.cn/brew.git
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://mirrors.ustc.edu.cn/homebrew-core.git
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: 16-core 64-bit zen3
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 GNU/Linux
OS: Ubuntu 22.04.2 LTS (jammy)
WSL: 2 (Microsoft Store)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: 3.0.2
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I use bash install brew but my fish can`t find the brew,only chsh -s /bin/bash can I use brew.

**Comment by SMillerDev** [comment]
Did you add it to your path?

**Reply by TwelveNight** [reply]
I try it follow the documents
I just copy this
```shell
test -d ~/.linuxbrew && eval "$(~/.linuxbrew/bin/brew shellenv)"
test -d /home/linuxbrew/.linuxbrew && eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
test -r ~/.bash_profile && echo "eval \"\$($(brew --prefix)/bin/brew shellenv)\"" >> ~/.bash_profile
echo "eval \"\$($(brew --prefix)/bin/brew shellenv)\"" >> ~/.profile
```
，but had no effect.
I also write this in my fish.config
```fish 
if test -d (brew --prefix)"/share/fish/completions"
    set -gx fish_complete_path $fish_complete_path (brew --'prefix)/share/fish/completions
end

if test -d (brew --prefix)"/share/fish/vendor_completions.d"
    set -gx fish_complete_path $fish_complete_path (brew --prefix)/share/fish/vendor_completions.d
end
```
It's also no effect.
I'm not good at linux,how can I use other methods to add it to my path?

**Reply by gromgit** [reply, marked as answer]
> I try it follow the documents I just copy this
> 
> ```shell
> test -d ~/.linuxbrew && eval "$(~/.linuxbrew/bin/brew shellenv)"
> test -d /home/linuxbrew/.linuxbrew && eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
> test -r ~/.bash_profile && echo "eval \"\$($(brew --prefix)/bin/brew shellenv)\"" >> ~/.bash_profile
> echo "eval \"\$($(brew --prefix)/bin/brew shellenv)\"" >> ~/.profile
> ```
> 
> ，but had no effect.

That just adds the necessary Homebrew configuration for *Bourne-like shells* (bash, zsh, etc.), not fish.

> I also write this in my fish.config
> 
> ```fish
> if test -d (brew --prefix)"/share/fish/completions"
>     set -gx fish_complete_path $fish_complete_path (brew --'prefix)/share/fish/completions
> end
> 
> if test -d (brew --prefix)"/share/fish/vendor_completions.d"
>     set -gx fish_complete_path $fish_complete_path (brew --prefix)/share/fish/vendor_completions.d
> end
> ```
> 
> It's also no effect.

That adds Homebrew shell completions for fish, *assuming the Homebrew configuration is already done*.

I don't use fish myself, but a quick look at the [fish configuration docs](https://fishshell.com/docs/current/index.html#configuration) suggests adding the following before the completions code above should work:
```fish
if status --is-login
    eval (/home/linuxbrew/.linuxbrew/bin/brew shellenv)
end
```
If that doesn't, then someone who actually uses fish regularly would need to advise further.

**Reply by TwelveNight** [reply]
Thank you,you answer is effective
